### PR TITLE
Clear for dialog fields

### PR DESCRIPTION
**Pull request by kaczymuczy**
## Description
Add the option to decide whether a dialog field should be cleared or not before adding new values to it. Currently only Multifield has the clear() method implemented, others are empty. More will be added later when required during our work on testing core components.
Part of https://github.com/Cognifide/bobcat/issues/373
Used by https://github.com/Cognifide/bobcat-aem-tests/pull/16

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] My code follows the [code styleguide](https://github.com/Cognifide/bobcat/blob/master/CONTRIBUTING.md#styleguide) of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: bb-aem-core/src/main/java/com/cognifide/qa/bb/aem/core/component/dialog/dialogfields/DialogField.java [hunk 1]
```diff
@@ -49,4 +50,9 @@
   default String getLabel() {
     return null;
   }
+
```
Comment (Shaihuludus) [inline bb-aem-core/src/main/java/com/cognifide/qa/bb/aem/core/component/dialog/dialogfields/DialogField.java]: Better solution would be default here. Thanks to this it wouldn't required empty methods and would not break any API

Comment (mkrzyzanowski) [reply bb-aem-core/src/main/java/com/cognifide/qa/bb/aem/core/component/dialog/dialogfields/DialogField.java]: Also, document the new behaviour please :)

Comment (kaczymuczy) [reply bb-aem-core/src/main/java/com/cognifide/qa/bb/aem/core/component/dialog/dialogfields/DialogField.java]: https://media1.tenor.com/images/c5d41bfa655c636286d80ae4552ce386/tenor.gif?itemid=10337157

Comment (kaczymuczy) [reply bb-aem-core/src/main/java/com/cognifide/qa/bb/aem/core/component/dialog/dialogfields/DialogField.java]: Fixed & fixed

File: docs/_docs/modules/aem/componentconfiguration.md [hunk 2]
```diff
@@ -96,6 +96,7 @@ Properties:
 - label - select dialog field with this label. If there is no label the fields will be selected in order.
 - type - type of dialog field. All available OOB fields are displayed in section below
 - value - what value should be set
+- clear - true/false, currently viable only for Multifields, you decide if the field's state should be set to a blank before entering the new value. This translates to deleting all existing items in a Multifield before adding any new ones.
```
Comment (mkrzyzanowski) [inline docs/_docs/modules/aem/componentconfiguration.md]: 👌

**Pull request merged**